std/csv.d

```diff
@@ -424,6 +366,101 @@ if (isInputRange!Range && is(Unqual!(ElementType!Range) == dchar)
         (input, cast(string[]) null, delimiter, quote);
 }
 
+
+/**
+The $(D Contents) of the input can be provided if all the records are the
+same type such as all integer data:
+*/
+@safe unittest
+{
+    import std.algorithm.comparison : equal;
+    string text = "76,26,22";
+    auto records = text.csvReader!int;
+    assert(records.equal!equal([
+        [76, 26, 22],
+    ]));
+}
+
+/**
+Using a struct with modified delimiter:
+*/
+@safe unittest
+{
+    import std.algorithm.comparison : equal;
+    string text = "Hello;65;2.5\nWorld;123;7.5";
+    struct Layout
+    {
+        string name;
+        int value;
+        double other;
+    }
+
+    auto records = text.csvReader!Layout(';');
+    assert(records.equal([
+        Layout("Hello", 65, 2.5),
+        Layout("World", 123, 7.5),
+    ]));
+}
+
+/**
+Specifying $(D ErrorLevel) as $(LREF Malformed.ignore) will lift restrictions
+on the format. This example shows that an exception is not thrown when
+finding a quote in a field not quoted.
+*/
+@safe unittest
+{
+    string text = "A \" is now part of the data";
+    auto records = text.csvReader!(string, Malformed.ignore);
+    auto record = records.front;
+
+    assert(record.front == text);
+}
+
+/// Read only column "b"
+@safe unittest
+{
+    import std.algorithm.comparison : equal;
+    string text = "a,b,c\nHello,65,63.63\nWorld,123,3673.562";
+    auto records = text.csvReader!int(["b"]);
+
+    assert(records.equal!equal([
+        [65],
+        [123],
+    ]));
+}
+
+/// Read from header of different order
```

Comment:
"Read while rearranging the columns by specifying a header with a different order"